### PR TITLE
Update test

### PR DESCRIPTION
**Pull request by dayhaysoos**
This PR is mostly for the updated tests for doing CRUD with DWN. I also added the Code of Conduct to this PR.

### CODE REVIEW

File: tests/node-esm/web5-test.js [hunk 1]
```diff
@@ -1,14 +1,82 @@
-import { Web5 } from '@tbd54566975/web5';
+import { Web5 } from "@tbd54566975/web5";
```
Comment (leordev) [inline tests/node-esm/web5-test.js:1]: Q: Why not take the opportunity to use the global dependency `@web5/api`?

File: tests/node-esm/web5-test.js [hunk 2]
```diff
@@ -1,14 +1,82 @@
...
+
+// const { record, status } = await web5.dwn.records.write({
+//   data: 'Hello!'
+// });
 
-console.log(JSON.stringify(status, null, 2));
-console.log(record);
\ No newline at end of file
+// console.log(JSON.stringify(status, null, 2));
+// console.log(record);
```
Comment (leordev) [inline tests/node-esm/web5-test.js:82]: I think you could throw away this vs just commenting

File: tests/node-esm/web5-test.js [hunk 3]
```diff
@@ -1,14 +1,82 @@
...
+
+  try {
+    result.web5 = await Web5.connect({
+      techPreview: {
+        dwnEndpoints: ["http://localhost:3000"],
+      },
+    });
+    did = result.web5.did;
+  } catch (error) {
+    console.log("Error:", error);
```
Comment (leordev) [inline tests/node-esm/web5-test.js:21]: nit: be descriptive about the error: `console.log("Web5.connect Error:", error);`

File: tests/node-esm/web5-test.js [hunk 4]
```diff
@@ -1,14 +1,82 @@
...
+  }
+
+  try {
+    const { record: _record } = await result.web5.dwn.records.write({
+      data: "Hello!",
+    });
+    result.record = _record;
+  } catch (error) {
+    console.log("Record Error:", error);
```
Comment (leordev) [inline tests/node-esm/web5-test.js:30]: nit: be descriptive about the error: `console.log("DWN.records.write Error:", error);`

File: tests/node-esm/web5-test.js [hunk 5]
```diff
@@ -1,14 +1,82 @@
...
 
-const { record, status } = await web5.dwn.records.write({
-  data: 'Hello!'
-});
+  console.log(JSON.stringify(result, null, 2));
+
+  return result;
+};
+
+checkWeb5();
```
Comment (leordev) [inline tests/node-esm/web5-test.js:69]: I love how you are not failing immediately on each operation, which means that we can test if read/update/delete are working regardless if any of them fail.

That being said, what would happen if connect or write fails? I think these two could abort the test early on.

Comment (leordev) [reply tests/node-esm/web5-test.js:69]: And here I think you can make assertions on each result object field. 

eg: you could collect error messages in a string array, like so:
- web5 is present else append "web5 is not present" to errors
- did starts with `did:ion:` and has at least N chars else append "invalid did" to errors
- record data is equal to whatever expected data and record id is a valid cid else...
- read status is equal to the success status code else...
- update status is equal to the success status code else...
- delete status is equal to the success status code else...

If errors.length  > 0: print errors messages and abort the process (exit code 1)
else: console log successful test!

File: tests/node-esm/web5-test.js [hunk 6]
```diff
@@ -1,14 +1,82 @@
...
-  techPreview: {
-    dwnEndpoints: ['http://localhost:3000']
+const checkWeb5 = async () => {
+  let result = {
+    web5: null,
+    did: null,
+    record: null,
+    readStatus: null,
+    updateStatus: null,
+    deleteStatus: null,
```
Comment (leordev) [inline tests/node-esm/web5-test.js:10]: suggestion: to be more thorough I'd probably add three read operations:
- after write (already exists)
- after update
- after delete

and have these records data in the results as well. this way you can compare if the data was updated as expected, and the data was deleted as well, aside from just checking the status.

maybe the lib threw a happy status but the data is not quite there.

File: tests/node-esm/web5-test.js [hunk 7]
```diff
@@ -1,14 +1,82 @@
...
+  }
+
+  try {
+    const { status } = await result.web5.dwn.records.delete({
+      recordId: record.id,
+    });
+
+    result.deleteStatus = status;
+  } catch (error) {
+    console.log("Delete Record Error:", error);
```
Comment (leordev) [inline tests/node-esm/web5-test.js:61]: another suggestion: on all the console log statements for error above, I'd change them to `console.error` vs using the `console.log`. Aside from having proper error logging, when running in our CI we would have nice errors summaries as well.

**Review by leordev — COMMENTED** [review]
I love to see this! It's a great start! I added a few suggestions/comments

File: tests/node-esm/web5-test.js (hunk 1 above)
Comment (dayhaysoos) [reply tests/node-esm/web5-test.js:1]: I think bc there are other DWN tests that will fail if I switch. I don't know how to replicate those other DWN tests, I guess that's something I wanted to ask you about but I'll bring it up in discord

File: tests/util/web5-test.mjs [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  } catch (error) {
+    console.error("Read Record Error:", error);
+  }
+
+  try {
+    const { status } = await result.record.update({ data: "Updated!" });
+    result.record.updateStatus = status;
+
+    const readStatus = (await result.record.data.text()) === "Updated!";
+    result.didUpdate = readStatus;
```
Comment (leordev) [inline tests/util/web5-test.mjs:50]: this readStatus is confusing. I think you meant to call this readStatus variable to didUpdate or just do a direct assignment in the `result.didUpdate`

File: tests/util/web5-test.mjs [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    const { status } = await result.record.delete();
+    result.deleteStatus = status.code;
+    result.didDelete = result.record.isDeleted;
+  } catch (error) {
+    console.error("Delete Record Error:", error);
+  }
+
+  console.log(result);
+
+  return result;
```
Comment (leordev) [inline tests/util/web5-test.mjs:65]: What do you think about doing the check here before returning the result?

eg: check all the statuses and variables are present and have the expected value, else throw an error.

**Review by leordev — APPROVED** [review]
Great work! LGTM 🎉 

Added a couple of comments for your consideration before merging!

File: tests/util/web5-test.mjs (hunk 8 above)
Comment (dayhaysoos) [reply tests/util/web5-test.mjs:50]: oh yeah all I wanted to do was compare text..not sure how I came up with that lol

File: tests/util/web5-test.mjs (hunk 9 above)
Comment (dayhaysoos) [reply tests/util/web5-test.mjs:65]: Sure, how do you think I should do the check? should I choose only a few things to check? Should I be checking by comparing another object?

**Pull request merged**